test/src/test/java/org/corfudb/integration/LogReplicationDynamicStreamIT.java

```diff
@@ -0,0 +1,388 @@
+     * (4) Open mapA at Sink and verify data replicated successfully
+     * (5) Write more entries to mapA at Source and verify log entry sync at Sink
+     */
+    @Test
+    public void testSnapshotAndLogEntrySync() throws Exception {
+        // Open mapA on Source
+        openMapAOnActive();
+
+        // writeToSource for initial snapshot sync
+        writeToMap(mapA, true, 0, numWrites);
+
+        // Confirm data does exist on Source Cluster
+        assertThat(mapA.count()).isEqualTo(numWrites);
+
+        startLogReplicatorServers();
+
+        // Verify snapshot sync complete by checking log replication status is log entry sync state
+        verifyInLogEntrySyncState();
+
+        // Open mapA on Sink after log replication started
+        openMapAOnStandby();
+
+        // Verify succeed of snapshot sync
+        verifyDataOnStandby(mapAStandby, numWrites);
+
+        // Add Delta's for Log Entry Sync
+        writeToMap(mapA, true, numWrites, numWrites / 2);
+
+        // Verify log replication status is log entry sync
+        verifyInLogEntrySyncState();
+
+        // Verify succeed of log entry sync
+        verifyDataOnStandby(mapAStandby, numWrites + (numWrites / 2));
+    }
+
+    /**
+     * We cannot get the list of streams to replicate in advance. This test will verify it works
+     * correctly when new streams are opened during log entry sync.
+     *
+     * (1) Perform basic snapshot and log entry sync, leave the clusters in log entry sync state
+     * (2) Open a new stream mapB at Source, write some entries to it
+     * (3) Verify at the Sink that mapB is replicated successfully during log entry sync
+     */
+    @Test
+    public void testNewStreamsInLogEntrySync() throws Exception {
+        // perform basic snapshot and log entry sync, the cluster should be in log entry sync state now
+        testSnapshotAndLogEntrySync();
+
+        // Verify log replication status is log entry sync
+        verifyInLogEntrySyncState();
+
+        String streamB = "Table002";
+        // open mapB at Source
+        Table<StringKey, IntValue, Metadata> mapB = corfuStoreActive.openTable(
+                NAMESPACE,
+                streamB,
+                StringKey.class,
+                IntValue.class,
+                Metadata.class,
+                TableOptions.fromProtoSchema(IntValueTag.class)
+        );
+
+        writeToMap(mapB, true, 0, numWrites);
+
+        // open mapB at Sink
+        Table<StringKey, IntValue, Metadata> mapBStandby = corfuStoreStandby.openTable(
+                NAMESPACE,
+                streamB,
+                StringKey.class,
+                IntValue.class,
+                Metadata.class,
+                TableOptions.fromProtoSchema(IntValueTag.class)
+        );
+
+        verifyDataOnStandby(mapBStandby, numWrites);
+    }
+
+    /**
+     * We need to clear local writes on Sink for streams with is_federated set to be true,
```

Comment:
This comment is confusing.  clear is only done for streams which evidenced data on the Sink but none from the Source.  Can you update the comment accordingly?

Reply:
Updated, thanks!